libavfilter/qsvvpp.c

```diff
@@ -300,10 +300,12 @@ static QSVFrame *submit_frame(QSVVPPContext *s, AVFilterLink *inlink, AVFrame *p
         qsv_frame->surface = (mfxFrameSurface1 *)qsv_frame->frame->data[3];
     } else {
         /* make a copy if the input is not padded as libmfx requires */
-        if (picref->height & 31 || picref->linesize[0] & 31) {
+        const AVPixFmtDescriptor *desc = av_pix_fmt_desc_get(picref->format);
+        if (picref->height & 31 || picref->linesize[0] & 31
+            || (!strcmp(desc->name, "yuv420p") && picref->linesize[0] & 63)) {
             qsv_frame->frame = ff_get_video_buffer(inlink,
-                                                   FFALIGN(inlink->w, 32),
-                                                   FFALIGN(inlink->h, 32));
+                                                   FFALIGN(inlink->w, 128),
+                                                   FFALIGN(inlink->h, 64));
```

Comment:
For the case of yuv420p:
1. picref->linesize[0] is checked as 64 bytes aligned, but here 128 bytes aligned is required for width. 
2.  picref->height hasn't been checked, but here 64 bytes aligned is required. 
3.  Is the 128x64 alignment requirement written in MSDK documentation?

Reply:
Hi,

I am not sure about the required alignment, it was in the code I had (I just added the check to enter in this case).

BR